### PR TITLE
windows: Refactor clipboard implementation

### PR DESCRIPTION
**Pull request by JunkuiZhang**
This PR provides a similar implementation to the macOS clipboard implementation, adds support for metadata and includes tests.

Release Notes:

- N/A

### CODE REVIEW

File: crates/gpui/src/platform/windows/platform.rs [hunk 1]
```diff
@@ -584,14 +602,6 @@ impl Platform for WindowsPlatform {
     }
 }
 
-impl Drop for WindowsPlatform {
```
Comment (mikayla-maki) [inline crates/gpui/src/platform/windows/platform.rs:587]: Why was this removed?

File: crates/gpui/src/platform/windows/platform.rs [hunk 2]
```diff
@@ -203,6 +222,7 @@ impl Platform for WindowsPlatform {
         if let Some(ref mut callback) = self.state.borrow_mut().callbacks.quit {
             callback();
         }
+        unsafe { OleUninitialize() };
```
Comment (JunkuiZhang) [inline crates/gpui/src/platform/windows/platform.rs:225]: `OleUninitialize()` hasn't been removed; it has been moved here. The reason for moving it is that running `OleUninitialize()` in `Drop` causes a `STATUS_ACCESS_VIOLATION` error during testing.

Comment (JunkuiZhang) [reply crates/gpui/src/platform/windows/platform.rs:225]: It's been fixed.

**Comment by JunkuiZhang** [comment]
After using `windbg` for in-depth analysis, I discovered that the issue is due to Rust's `Drop` order. When `WindowsPlatform` is deconstructed, it first runs its own `Drop` method, and then the `Drop` methods of its member variables. This means `OleUninitialize()` is called first, followed by the `Drop` method for `DirectWriteTextSystem`. The `DirectWriteTextSystem` contains an `IWICImagingFactory2` that relies on the `Com` environment. Therefore, it also needs the `Com` environment during its deconstruction. However, since the `Com` environment no longer exists due to the earlier call to `OleUninitialize()`, a `STATUS_ACCESS_VIOLATION` error occurs. Strangely, this error only appears during testing, and not in normal builds.

**Comment by mikayla-maki** [comment]
Thanks!
